apps/ios/Shared/Model/SimpleXAPI.swift

```diff
@@ -341,6 +341,11 @@ func loadChat(chat: Chat, search: String = "") {
         im.reversedChatItems = []
         let chat = try apiGetChat(type: cInfo.chatType, id: cInfo.apiId, search: search)
         m.updateChatInfo(chat.chatInfo)
+        if case let .direct(contact) = chat.chatInfo, cInfo.chatDeleted == false, chat.chatInfo.chatDeleted {
+            var updatedContact = contact
+            updatedContact.chatDeleted = false
+            m.updateContact(updatedContact)
+        }
```

Comment:
probably not both are needed - we only need to update chat once, right? so either updateChatInfo or updateContact, smth like this:
```suggestion
        if case let .direct(contact) = chat.chatInfo, cInfo.chatDeleted == false, chat.chatInfo.chatDeleted {
            var updatedContact = contact
            updatedContact.chatDeleted = false
            m.updateContact(updatedContact)
        } else {
            m.updateChatInfo(chat.chatInfo)
        }
```